packages/service-library/src/servicelib/fastapi/openapi.py

```diff
@@ -166,3 +166,37 @@ def _custom_openapi_method(self: FastAPI) -> dict:
         return output
 
     setattr(app, "openapi", types.MethodType(_custom_openapi_method, app))  # noqa: B010
+
+
+def create_openapi_specs(
+    app: FastAPI,
+    *,
+    drop_fastapi_default_422: bool = True,
+    remove_main_sections: bool = True,
+):
+    """
+    Includes some patches used in the api/specs generators
+    """
+    override_fastapi_openapi_method(app)
+    openapi = app.openapi()
+
+    # Remove these sections
+    if remove_main_sections:
+        for section in ("info", "openapi"):
+            openapi.pop(section, None)
+
+    schemas = openapi["components"]["schemas"]
+    for section in ("HTTPValidationError", "ValidationError"):
+        schemas.pop(section, None)
+
+    # Removes default response 422
+    if drop_fastapi_default_422:
+        for method_item in openapi.get("paths", {}).values():
+            for param in method_item.values():
+                # NOTE: If description is like this,
+                # it assumes it is the default HTTPValidationError from fastapi
+                if (e422 := param.get("responses", {}).get("422", None)) and e422.get(
+                    "description"
+                ) == "Validation Error":
+                    param.get("responses", {}).pop("422", None)
```

Comment:
what is this?

Reply:
This utility customizes the default behavior of FastAPI for generating OpenAPI schemas related to 422 status code errors (e.g., validation errors).

Its primary use is within the OpenAPI generators located in `api/specs`. It's important to note that the storage and webserver components do not handle 422 errors in the same way as FastAPI. Therefore, it's necessary to exclude these errors from the OpenAPI schemas. At the time of implementation, FastAPI did not provide a native mechanism to achieve this.